Tests/test_SeqIO_index.py

```diff
@@ -49,8 +48,6 @@ def gzip_open(filename, format):
     # At time of writing, under Python 3.2.2 seems gzip.open(filename, mode)
     # insists on giving byte strings (i.e. binary mode)
     # See http://bugs.python.org/issue13989
-    if sys.version_info[0] < 3 or format in SeqIO._BinaryFormats:
```

Comment:
This would now become:

```python
if format in SeqIO._BinaryFormats:
```

If the tests passed anyway this needs closer examination....
```

Reply:
It passed all the tests, should I  add `if format in SeqIO._BinaryFormats:` ?

Reply:
The tests passed as is (removing the if statement and special handling of binary files). That bothers me - I would like to know why it didn't fail.

Reply:
okay, I want to know what should be the behavior of this code, please let me know what can I do to help here?

Reply:
In Python 3, ``gzip.open()`` defaults to a binary (bytes) handle, but ``open()`` defaults to a text (unicode) handle.

The SeqIO indexing (e.g. ``SeqIO.index``) takes a filename not a handle, and always opens the handle itself using binary (bytes) mode.

Most of the SeqIO parsers (via ``SeqIO.parse``) expect a text mode (unicode) handle, except the binary formats like SFF which expect a binary (bytes) handle.

I suggest you exclude this file from this pull request.

Reply:
Digging into this more, the SFF code needs more careful review for Python 3 only - it currently support Python 2 in non-binary mode, which it no longer relevant.

Reply:
I'm going to open a separate pull request for SFF and its tests.

The answer to the immediate question, we didn't use a GZIP (or BGZF) compressed binary file in the tests invoking this ``gzip_open`` function, so the code path was not used.

Reply:
See #2499.

Reply:
I have reset the file to its original and I am seeing your code for fixing the problem, thank you for describing the problem and providing the code in #2499.